### ISSUE TITLE
AtlasTexture does not repeated in sprites

### ISSUE THREAD

**Issue by rlegonidec** [godotengine/godot#5660]
**Operating system or device - Godot version:**
Godot Engine 2.0.4.1 (Ubuntu 16.04 - 64bits)

**Issue description** (what happened, and what was expected):
The AtlasTexture is not repeated on the sprite.

**Steps to reproduce:**
* Create a Sprite
* Load the AtlasTexture (or create a new one)
* Check Region on and set Region Rect (0, 0, 300, 300) (to have the "repeated effect")
* Click Edit the Texture
* Click Edit the Atlas
* Check Repeat on ImageTexture Flags.

**Image example:**

![sprite_atlastexture_bug](https://cloud.githubusercontent.com/assets/1983402/16783300/c296a5a2-4883-11e6-8f62-8d23c073513d.png)

**Comment by ret80** [comment]
this is a normal situation. it should be

**Comment by bojidar-bg** [comment]
Well, even if it is normal, it would be nice if it is either fixed or explained in the docs.

**Comment by rlegonidec** [comment]
Can you explain why the AtlasTexture does that ?
I use many AtlasTexture instead of ImageTexture. In this case what is the difference ?

**Comment by ret80** [comment]
I'm afraid my English it would be hard to do, but I'll try.
The fact that the repetition of the texture makes the graphics api itself. we just do the settings, specify the texture coordinates is greater than 1 and that the texture must be repeated. In the case of an atlas, you can repeat the whole atlas of the whole, but a separate part is unlikely.
In your case, just do not need to add the textures in the atlas that you want to display in this way. I think that these textures you will not have a lot, and it's not like will not affect the performance.

**Comment by bojidar-bg** [comment]
@godotengine/bugsquad: you might consider retagging as <kbd>documentation</kbd> if this is so indeed.

**Comment by akien-mga** [comment]
@bojidar-bg You're in bugsquad too :)

**Comment by SuperUserNameMan** [comment]
why would one want to apply a GL_REPEAT to texture atlas ?

the GL_REPEAT applies only when the texture UV mapping exceed the borders of the texture.
AFAIUC*, atlas are used to gather multiple textures into a single one. So, when displaying one of them, it is not possible to set texture UV coords that exceed the borders of the texture-atlas. GL_REPEAT cannot apply.

(*As Far As I Understand Correctly)

**Comment by bojidar-bg** [comment]
@SuperUserNameMan AFAIUC it is so indeed, but I would like a confirmation by either @reduz or someone else knowledgeable.

**Comment by rlegonidec** [comment]
Ok, I understand better why "it is normal" and it is OpenGL that repeats the entire texture despite the UV.

I have tested with a simple ImageTexture, and the rendering isn't good.
Unlike my atlas, the size of the image is not power of 2.
The repeat option seems not work without MipMaps. And the Mipmaps add a "blur effect".
The AtlasTexture's rendering matched exactly with the Inkscape's export.

How to correctly repeat a "not in power of 2" image in a sprite ?

![dessin](https://cloud.githubusercontent.com/assets/1983402/16845325/c74ae66a-49e8-11e6-8411-a8b68651c180.png)

**Comment by DualMatrix** [comment]
So this is not a bug but just a documentation issue right?

**Comment by OvermindDL1** [comment]
@DualMatrix Yep.  You can work around it by making a shader and passing in the uv extents of the rect so the shader itself can 'wrap' just the required rect instead of the whole image.

**Comment by cjmxp** [comment]
This bug now exists.

**Comment by Mickeon** [comment]
Th is no longer the case in `master`.
![image](https://user-images.githubusercontent.com/66727710/192233065-ab39f1fe-5d35-45fc-82a2-25dee44fd66f.png)

The documentation has yet to be updated, however.
https://github.com/godotengine/godot/blob/3472bdd6b6dd7c085b716caabdda16f88e6f167e/doc/classes/AtlasTexture.xml#L8

**Comment by KoBeWi** [comment]
> This is no longer the case in master.

Looks like it doesn't work again and the documentation implies that only nested AtlasTextures won't repeat.
